src/apispec/ext/marshmallow/openapi.py

```diff
@@ -147,105 +141,38 @@ def schema2parameters(
 
         fields = get_fields(schema, exclude_dump_only=True)
 
-        return self.fields2parameters(fields, default_in=default_in)
-
-    def fields2parameters(self, fields, *, default_in):
-        """Return an array of OpenAPI parameters given a mapping between field names and
-        :class:`Field <marshmallow.Field>` objects. If `default_in` is "body", then return an array
-        of a single parameter; else return an array of a parameter for each included field in
-        the :class:`Schema <marshmallow.Schema>`.
-
-        In OpenAPI3, only "query", "header", "path" or "cookie" are allowed for the location
-        of parameters. In OpenAPI 3, "requestBody" is used when fields are in the body.
-
-        This function always returns a list, with a parameter
-        for each included field in the :class:`Schema <marshmallow.Schema>`.
-
-        https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#parameterObject
-        """
-        parameters = []
-        body_param = None
-        for field_name, field_obj in fields.items():
-            if field_obj.dump_only:
-                continue
-            param = self.field2parameter(
+        return [
+            self._field2parameter(
                 field_obj,
                 name=self._observed_name(field_obj, field_name),
```

Comment:
This is probably separate but we could make `self._observed_name` a function. It's not really doing much as a static method. That would clean this up a little in my opinion.

Reply:
This method is removed in #583 anyway.